### PR TITLE
Layout fixes

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

This PR contains fixes for several bugs found by me and @Frizi during development of our current tasks. Created unit tests for each case (and many more).

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by farmaazon** [comment]
Created as draft, as I still need to test if changes does not break something in IDE. Besides, it's ready for review.

File: lib/rust/ensogl/core/src/display/object/instance.rs [hunk 1]
```diff
@@ -3058,20 +3059,29 @@ where
...
     fn reset_size_to_static_values<Dim>(&self, x: Dim, parent_size: f32)
     where Dim: ResolutionDim {
         let size = match self.layout.size.get_dim(x) {
             Size::Fixed(unit) => unit.resolve_const_and_percent(parent_size).unwrap_or(0.0),
-            Size::Hug => 0.0,
+            Size::Hug
+                if self.dirty.transformation.check()
+                    || self.dirty.modified_children.check_all()
+                    || self.dirty.removed_children.check_all() =>
+                0.0,
```
Comment (wdanilo) [inline lib/rust/ensogl/core/src/display/object/instance.rs]: Please check if it is possible to minimize the amount of checks, as we agreed via phone.

File: lib/rust/ensogl/core/src/display/object/instance.rs [hunk 2]
```diff
@@ -3058,20 +3059,29 @@ where
...
         self.layout.computed_size.set_dim(x, size);
     }
 
     /// The main entry point from the recursive auto-layout algorithm.
     fn refresh_layout_internal(&self, pass: LayoutResolutionPass, pass_cfg: PassConfig) {
-        if self.dirty.transformation.check() || self.dirty.modified_children.check_all() {
+        if self.dirty.transformation.check()
+            || self.dirty.modified_children.check_all()
+            || self.dirty.removed_children.check_all()
+        {
```
Comment (wdanilo) [inline lib/rust/ensogl/core/src/display/object/instance.rs]: could you please refactor that to one helper fn?

**Review by wdanilo — APPROVED** [review]

**Review by wdanilo — APPROVED** [review]
@farmaazon and @Frizi amazing job with debugging and fixing it! I'm sorry I did not spot these issues when I was developing the code.

**Comment by wdanilo** [comment]
Merged, as we need it to develop the codebase, and the CI problems seem unrelated.

**Pull request merged**